### ISSUE TITLE
Product changes can't be published when catalog is not turned on

### ISSUE THREAD

**Issue by prinzdezibel** [reactioncommerce/reaction#4001]
NOTE: This ticket could be seen as a duplicate of #3978 , but I think it's not. This one is not so much dealing with the eye-ball icon / publish button, but with the general way how the product grid retrieves the data from the catalog.

## Prerequisites
* [x] Are you running the latest version?
* [x] Are you able to consistently reproduce the issue?
* [x] Did you search the issue queue for existing issue? 

## Issue Description
I would expect changes to be published implicitly (with no need to press a publish button or whatsoever),  when the product revisions are turned off (not active).

This is not the case. Also there's no button to publish the changes manually (which I personally would find awkward if there would be one, btw).

## Steps to Reproduce
1. As an admin, create and publish a product with product revisions turned on.
2. Disable product revision
3. Change product properties (e.g. title)
4. Visit the page as user and observe that those changes are not published.

### Versions
Node: 8.2.1
NPM: 5.3.0
Meteor Node: 8.9.4
Meteor NPM: 5.6.0
Reaction CLI: 0.26.4
Reaction: 1.9.0
Reaction branch: release-1.9.0
Docker: 17.03.1-ce

**Comment by prinzdezibel** [comment]
Certainly a regression, but would need to verify to be 100% sure.

**Comment by zenweasel** [comment]
Can you clarify this ticket? Catalog is not something that gets turned on or off. You will always need to publish things to the catalog whether you have revisions turned on or off when you want them to be visible to consumers.

**Comment by mikemurray** [comment]
@prinzdezibel If revision control was disabled, you'd still have to publish your changes to the catalog. The catalog acts as a denormalized collection of products for customer viewing.

I think the issue is that the publish button should not be hidden when revision control is disabled.

**Comment by zenweasel** [comment]
@mikemurray I feel like this process needs a little work? Now we need to make products visible, publish them and publish them again and it's not clear where a particular product is in that workflow from the PDP?

**Comment by mikemurray** [comment]
@zenweasel publishing them twice?

I think PDP would eventually use the catalog collection for customers. While we use the products collection for administration.

**Comment by zenweasel** [comment]
I mean, I guess why do revisions exist if we have the catalog?

**Comment by mikemurray** [comment]
Exactly! Revision control is redundant to the Catalog collection in its current form as they both allow you to stage your changes before going out the customers. At the moment, there no way in the UI to go back to a previous revision, only discard your current changes.

**Comment by spencern** [comment]
Right now, the PDP does not pull data from the catalog. That's an update we'll need to make before we can really deprecate revisions.

**Comment by zenweasel** [comment]
When would we want to pull data from the catalog into the PDP? Isn't the version in "products" the canonical one?

**Comment by zenweasel** [comment]
So despite my digression here, this is a valid issue for "don't hide the publish button" when revisions is turned off otherwise you can't publish to catalog?

**Comment by mikemurray** [comment]
@zenweasel correct, we should not hide the publish button. The publish controls visibility should no longer be tied to whether or not revisions is enabled.

**Comment by prinzdezibel** [comment]
I understand. Then my expectations were wrong when I said:

> I would expect changes to be published implicitly (with no need to press a publish button or whatsoever), when the product revisions are turned off (not active).

And the missing publish button is indeed covered by #3978. Closing this issue in favour of #3978. Feel free to reopen if you feel there's more to it.
